nostr/relay.py

```diff
@@ -7,42 +8,52 @@
 from .message_type import RelayMessageType
 from .subscription import Subscription
 
+@dataclass
 class RelayPolicy:
-    def __init__(self, should_read: bool=True, should_write: bool=True) -> None:
-        self.should_read = should_read
-        self.should_write = should_write
-
+    should_read: bool = True
+    should_write: bool = True
+    
     def to_json_object(self) -> dict[str, bool]:
         return { 
             "read": self.should_read, 
             "write": self.should_write
         }
 
+
+
+@dataclass
+class RelayProxyConnectionConfig:
+    host: str
+    port: int
+    type: str
+
+
+
+@dataclass
 class Relay:
-    def __init__(
-            self, 
-            url: str, 
-            policy: RelayPolicy, 
-            message_pool: MessagePool,
-            subscriptions: dict[str, Subscription]={}) -> None:
-        self.url = url
-        self.policy = policy
-        self.message_pool = message_pool
-        self.subscriptions = subscriptions
-        self.lock = Lock()
-        self.ws = WebSocketApp(
-            url,
+    url: str
+    message_pool: MessagePool
+    policy: RelayPolicy = RelayPolicy()
+    ssl_options: dict = None
+    proxy_config: RelayProxyConnectionConfig = None
+
+    def __post_init__(self):
+        self.subscriptions: dict[str, Subscription] = {}
+        self.lock: Lock = Lock()
+        self.ws: WebSocketApp = WebSocketApp(
+            self.url,
             on_open=self._on_open,
             on_message=self._on_message,
             on_error=self._on_error,
-            on_close=self._on_close)
+            on_close=self._on_close
+        )
 
-    def connect(self, ssl_options: dict=None, proxy: dict=None):
+    def connect(self):
         self.ws.run_forever(
-            sslopt=ssl_options,
-            http_proxy_host=None if proxy is None else proxy.get('host'), 
-            http_proxy_port=None if proxy is None else proxy.get('port'),
-            proxy_type=None if proxy is None else proxy.get('type')
+            sslopt=self.ssl_options,
+            http_proxy_host=None if self.proxy_config is None else self.proxy_config.host, 
```

Comment:
A proper syntax os this would be something like

```suggestion
            http_proxy_host=self.proxy_config.host, 
```

where the attribute `.host` is `None` by default.